### PR TITLE
feat: Snooze and mark as pending conversation

### PR DESCRIPTION
**Pull request by iamsivin**
# Pull Request Template

## Description 

- Snooze conversation
- Mark as pending conversation
- Adds an icon in the chat header if the conversation is pending or snoozed
- Moved UIkitten theming to v2 in the chat header
- Set white color for status bar 

Fixes https://linear.app/chatwoot/issue/CW-1272/snooze-conversation

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

**Screenshot**
image

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have tested in both Android and iOS platform
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/screens/ChatScreen/components/ChatHeader.js [hunk 1]
```diff
@@ -325,23 +421,28 @@ const ChatHeader = ({
...
+            <View style={styles.titleView}>
+              <View style={styles.customerName}>
                 {customerDetails.name && (
-                  <CustomText style={style.headerTitle}>
-                    {customerDetails.name && customerDetails.name.length > 24
-                      ? ` ${customerDetails.name.substring(0, 20)}...`
-                      : ` ${customerDetails.name}`}
-                  </CustomText>
+                  <Text md medium color={colors.textDark}>
+                    {getTextSubstringWithEllipsis(customerDetails.name, 12)}
```
Comment (iamsivin) [inline src/screens/ChatScreen/components/ChatHeader.js:428]:
```suggestion
                    {getTextSubstringWithEllipsis(customerDetails.name, 14)}
```

File: src/components/ConversationActionItem.js [hunk 2]
```diff
@@ -81,13 +86,13 @@ const ConversationActionItem = ({
           </Text>
         </View>
         <View style={styles.sectionActionView}>
-          {itemType === 'assignee' && thumbnail !== '' && (
+          {shouldShowUserAvatar && (
             <UserAvatar thumbnail={thumbnail} userName={name} size={18} fontSize={8} />
```
Comment (muhsin-k) [inline src/components/ConversationActionItem.js:90]: The user avatar is not loading.

File: src/screens/ChatScreen/components/ChatHeader.js [hunk 3]
```diff
@@ -1,85 +1,99 @@
 import React, { useRef, useCallback, useMemo } from 'react';
-import {
-  withStyles,
-  Icon,
-  TopNavigation,
-  TopNavigationAction,
-  Spinner,
-} from '@ui-kitten/components';
-import BottomSheetModal from 'components/BottomSheet/BottomSheet';
+import { useTheme } from '@react-navigation/native';
```
Comment (muhsin-k) [inline src/screens/ChatScreen/components/ChatHeader.js:2]: This component is very big. Can we split it into multiple components?

File: src/screens/ChatScreen/components/ChatHeader.js [hunk 4]
```diff
@@ -1,85 +1,99 @@
...
+import { View, Share, ActivityIndicator, Dimensions, StyleSheet } from 'react-native';
 import { getTypingUsersText, getCustomerDetails } from 'helpers';
-import CustomText from 'components/Text';
 import { selectConversationToggleStatus } from 'reducer/conversationSlice';
 import conversationActions from 'reducer/conversationSlice.action';
-import { UserAvatar, Pressable } from 'components';
+import { UserAvatar, Pressable, Text, Icon } from 'components';
 import { getInboxName } from 'helpers';
 import Banner from 'src/screens/ChatScreen/components/Banner';
 import InboxName from 'src/screens/ChatScreen/components/InboxName';
```
Comment (muhsin-k) [inline src/screens/ChatScreen/components/ChatHeader.js:13]: User absolute path every where

File: src/screens/ChatScreen/components/ChatHeader.js [hunk 5]
```diff
@@ -1,85 +1,99 @@
...
 import TypingStatus from 'src/screens/ChatScreen/components/UserTypingStatus';
```
Comment (muhsin-k) [inline src/screens/ChatScreen/components/ChatHeader.js:14]:
```suggestion
import TypingStatus from 'screens/ChatScreen/components/UserTypingStatus';
```

File: src/screens/ChatScreen/components/ChatHeader.js (hunk 3 above)
Comment (iamsivin) [reply src/screens/ChatScreen/components/ChatHeader.js:2]: @muhsin-k Sure I can pick it as a separate task. If its fine we can merge this PR. Its open for a while.

File: src/components/ConversationActionItem.js (hunk 2 above)
Comment (iamsivin) [reply src/components/ConversationActionItem.js:90]: @muhsin-k Loading for me
image

File: src/screens/ChatScreen/components/ChatHeader.js (hunk 3 above)
Comment (iamsivin) [reply src/screens/ChatScreen/components/ChatHeader.js:2]: @muhsin-k Made some changes in reducing code lines. https://github.com/chatwoot/chatwoot-mobile-app/pull/666/commits/1e2f352b9fdef2c86d9f0af871a275b905f218cb
(reduced 100 lines)

**Pull request merged**